server/src/services/commits.ts

```diff
@@ -94,11 +94,23 @@ const addCommit = async (
     );
   }
 
-  return await commitStorage.addCommit({
+  return commitStorage.addCommit({
     ...DEFAULT_COMMIT_PAYLOAD,
     ...commitData,
     createdAt: new Date(),
   });
 };
 
-export default {getAllCommits, addCommit};
+/**
+ * Deletes the commit with the specified commitId.
+ * An error would be thrown if commit does not already exist, and if deletion fails.
+ * @param commitId Id of the commit to be deleted
+ */
+const deleteCommit = async (commitId: number) => {
+  const commit = await commitStorage.getCommit(commitId);
+  const commitListing = await listingStorage.getListing(commit.listingId);
+
+  return commitStorage.deleteCommit(commitId, commitListing.id);
```

Comment:
Why do you have to get the listing id in this layer? This feels more like a storage thing than a logic thing. It's not really a business logic that require you to retrieve the commit object and get the related listing id, but more related to the way these data are stored.

Reply:
@karenfrilya97 Hmm I agree it felt abit odd passing both commitId and listingId in as arguments, but I also don't really want to fetch commit twice, both in `services` layer and in `storage` layer. (I need to get in `services` layer because I want to do biz logic checks on commitStatus) One way is to pass the `commit` object to the `storage` layer, but that's weird too because this `commit` object is a `CommitResponse` object

Reply:
@hzxcaryn Ohh! I didn't know it's for `commitStatus` checking. In that case that's fine!
I personally think passing `commit` object looks nice (because it's something like 'please delete this commit'), but I also don't like the idea of passing more than what is necessary. I guess this is fine for now. Maybe if we can think of a better way to do it in the future we can change it.